lib_test/test_mustache.ml

```diff
@@ -103,6 +103,15 @@ let tests = [
     , [ ( `O [ "a" , `String "foo" ],
           "foo" ) ] ) ;
 
+    ("Begin\n{{#foo}} {{#bar}}\nMiddle\n{{/bar}} {{/foo}}\nEnd\n"
+    , concat [
+        raw "Begin\n";
+        section ["foo"] (section ["bar"] (raw "Middle\n"));
+        raw "End\n";
+      ]
+    , [ ( `O [ "foo" , `O []; "bar", `O [] ],
```

Comment:
A cram test would make it easier to demonstrate the fix.

Reply:
There is a difficulty in terms of test organization, though: if we want to split the binary as a separate package, I probably shouldn't write a cram test using the binary in the library tests, as this would introduce a suspicious circular-like dependency (lib{test} -> bin -> lib). (For now the cram tests in `bin/test` are more like integration tests for the binary behavior, rather than arbitrary spec tests.)

I could write a cram test using the library only. For now I went the easy way of just using 0-indented string literals with real newlines instead of `\n` escapes, to make the current test easier to read and understand.